enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,373 @@
+frequently and we can mitigate them on a case-by-case basis.
+
+
+## Design Details
+
+### Identified Overlap Points for Agent Installer
+
+During the discussions with the Metal Platform team it was identified that the following
+components of dev-scripts are the likely points of overlap for the new agent installer project:
+
+* The [01_install_requirements.sh](https://github.com/openshift-metal3/dev-scripts/blob/master/01_install_requirements.sh) 
+and [02_configure_host.sh](https://github.com/openshift-metal3/dev-scripts/blob/master/02_configure_host.sh) 
+scripts that are run using `make`.
+* Ansible playbooks to set up VMs such as: [vm_setup_vars.yml](https://github.com/openshift-metal3/dev-scripts/blob/master/vm_setup_vars.yml)
+
+These will be the things that we will focus on first when refactoring parts of dev-scripts.
+
+### Overall Components of the e2e Framework
+
+Using the general flow of an e2e test we can sort the overall components of the e2e framework.
+A general flow of an e2e test has the following phases:
+
+* Pre Environment Setup: 
+** Setting up VMs, networks, locations of images, image registries, etc
+* Build: 
+** Building the agent-based installer ISO and container images associated with it
+* Test:
+**  Test scenarios with pass/failure outputs
+* Reporting:
+** Reports comprised of test results
+* Environment Teardown
+** Removing VMs, networks, etc to start fresh again
+
+#### Pre Environment Setup 
+
+* The e2e framework will be a collection of Ansible playbooks and roles
+* Using roles to define reusable steps scoped to units of work
+* Use playbooks to set up common e2e workflows and chains for the project
+
+In discussions regarding this enhancement we identified the dev-scripts has components that are
+a part of the Pre Environment Setup and Environment Teardown phases of the framework. We will
+move forward with adding and potentially refactoring scripts in this repository for these phases.
+
+#### Build
+
+The scripts/automation for building the agent-based installer should be inside the repository
+where the installer is being worked on and not dev-scripts. We will reference the built 
+container images and ISO in dev-scripts from those repositories. Currently this project is being 
+worked on in the  openshift-agent-team/fleeting repository but this may change in the future.
+
+#### Deployment
+
+Deploying the agent-based installer will be heavily linked to the environment set up. This
+logic will land in dev-scripts alongside the pre environment setup steps.
+
+#### Tests
+
+For the actual test scenarios themselves we should use Ginkgo/Gomega. This is 
+similar to the rest of Openshift. Well most of (if not all) the product is written in Golang
+so it makes sense to use a Golang framework for these.
+
+There is a repository created as a preliminary place to put these. Plan to use a similar
+structure to openshift-tests in this repository.
```

Comment:
Would the long-term plan be to move this to openshift-tests once we move the code being tested into the openshift/ org?

Reply:
I don't see any particular blocking reason for not adopting it since the very beginning. Moreover, `openshift-tests` can be easily used in dev environment as a test runner.

Reply:
If we can just plug directly into `openshift-tests` this seems like a good plan to me.